logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/Message.kt

```diff
@@ -304,8 +305,17 @@ sealed interface Message {
         }
     }
 
-    enum class Status {
-        PENDING, SENT, READ, FAILED, FAILED_REMOTELY
+    sealed class Status {
```

Comment:
```suggestion
    sealed interface Status {
```

What's the benefit of using a sealed type instead of `enum` for this case? In any case, I'd suggest using sealed `interface` over `class` for extra flexibility and less `()`

Reply:
it was still in draft, but good suggestion anyways I created sealed class because of adding the counter inside the read data class, I am open to suggestions though, but I think it makes most sense having the structure like this inside the domain object, because the alternative would be to have it as additional field, but readCount does not make sense in any other state

Reply:
I changed it back to class in order to override toString(), is there any other way to still use interface and have the option to override toString() ? I guess that would be done inside each "status" so it will end up having more code

Reply:
> I changed it back to class in order to override toString(), is there any other way to still use interface and have the option to override toString() ? I guess that would be done inside each "status" so it will end up having more code

Yes. It would be in each implementation.

I'm also exploring Kotlin 1.9 on Kalium, so we can use `data object` and this can be solved :)

Reply:
awesome, I did not know such thing will be available !